### DISCUSSION TITLE
issue: When using `FieldPathByValue` with `useController` and `setValue` from `useFormContext`, the types should link up

### DISCUSSION THREAD

**Discussion by Svish** [react-hook-form/react-hook-form#9014] [Q&A]
### Version Number

7.35.0-next.0

### Codesandbox/Expo snack

https://codesandbox.io/s/broken-bash-i01f0y?file=/src/StringListInput.tsx

### Steps to reproduce

1. Make a custom input component using `useController` and `useFormContext`
2. Use the new `FieldPathByValue` for the `name`
3. Try to use `value` and `setValue`, but `value` is then `any`, and `setValue` seems confused?

### Expected behaviour

Given `name` defined as `FieldPathByValue<MyFieldValues, string>`:

* The `value` coming from `useController({ name, control })` should be of type `string` (or possibly `string | undefined`, in case of no default?). It's currently `any`.
* The `value` parameter of `setValue` from `useFormContext`, should should be `string`, when called with the `name` restricted to `string` values.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by bluebill1049** [comment]
thanks for the issue report, this is something we can improve later on as this is currently not the scope of the type update. However, happy to hear suggest or ideas on how can we get there.

**Reply by Svish** [reply]
Yeah, sorry, I guess this should be more like a feature request, than an issue/bug. The current scope is a great addition in itself, as it makes the DX when consuming custom components using `useController` potentially much better. 👍

As for "how to get there", I'm unfortunately not familiar enough with `react-hook-form` or experienced enough with TS, so not sure I can contribute much. I did take a look at the underlaying type for e.g. `setValue`, and it makes sense to me, but I don't know how one could "link it up" correctly, in a way that makes sense in a generic context like this.

**Reply by radoslavkarlik** [reply]
Any update on this?
